### PR TITLE
time integrated flux diagnostics

### PR DESCRIPTION
**Pull request by rgknox**
### Description:

This adds a site-level diagnostic that compares the total amount of mass in live vegetation (plants plus seeds) to the total integrated fluxes in and out of that pool over the life of the simulation.  The same is true for the litter pool. History output for the bias on this comparison is added as well, to see if the bias grows over time.

### Collaborators:

@JessicaNeedham 

### Expectation of Answer Changes:

This **does** have a bug fix, so some diagnostics related to litter fluxes may be corrected.

### Checklist

*If this is your first time contributing, please read the [**CONTRIBUTING**](https://github.com/NGEET/fates/blob/main/CONTRIBUTING.md) document.*

All checklist items must be checked to enable merging this pull request:

*Contributor*
- [ ] The in-code documentation has been updated with descriptive comments
- [ ] The documentation has been assessed to determine if updates are necessary

*Integrator*
- [ ] FATES PASS/FAIL regression tests were run
- [ ] Evaluation of test results for answer changes was performed and results provided

### Documentation

- [Technical Note](https://github.com/NGEET/fates-docs) update:
- [User's Guide](https://github.com/NGEET/fates-users-guide) update: 

### Test Results:

*CTSM (or) E3SM (specify which) test hash-tag:*

*CTSM (or) E3SM (specify which) baseline hash-tag:*

*FATES baseline hash-tag:*

*Test Output:*

### CODE REVIEW

**Comment by rgknox** [comment]
The bias between integrated flux and live-vegetation and litter states are looking pretty good now, see plots below, the scale on liveveg and litter is e-8 and e-14 respectively. 

I had to move the calculation of the flux integration to later in the call sequence so that termination mortality that occurs after vegetation dynamics is encapsulated.  I'll also run a test with CNP to see if this was the cause of #1218 .


vegc
interr_litter
interr_liveveg

**Review by mpaiao — COMMENTED** [review]
Thanks for addressing the mass conservation assessment, @rgknox ! I went through the code and it looks great, I just had a couple of minor clarification questions and suggestions.

File: main/ChecksBalancesMod.F90 [hunk 1]
```diff
@@ -240,7 +243,134 @@ subroutine CheckLitterPools(currentSite,bc_in)
...
+
+            select case(element_list(el))
+            case(carbon12_element)
+               net_uptake = diag%npp + site_mass%net_root_uptake*area_inv
+            case(nitrogen_element)
+               net_uptake = site_mass%net_root_uptake*area_inv
+            case(phosphorus_element)
+               net_uptake = site_mass%net_root_uptake*area_inv
+            case default
+               write(fates_log(),*) 'FATES: Invalid choice for cohort_fusion_conservation_method'
```
Comment (JessicaNeedham) [inline main/ChecksBalancesMod.F90:305]: Looks like the message to FATES log needs updating. 'FATES: Invalid element in CheckIntergratedMassPools'

File: main/ChecksBalancesMod.F90 [hunk 2]
```diff
@@ -240,7 +243,134 @@ subroutine CheckLitterPools(currentSite,bc_in)
...
+
+            ediag%err_liveveg = ibal%iflux_liveveg - ibal%state_liveveg
+
+            ediag%err_litter  = ibal%iflux_litter - ibal%state_litter
+
+
+            
+            !print*, ediag%err_liveveg!, ediag%err_liveveg/ibal%state_liveveg, ibal%state_liveveg, net_uptake, tot_litter_input
+
+            ! Perform the comparison between integrated flux and state
```
Comment (JessicaNeedham) [inline main/ChecksBalancesMod.F90:339]: I think we discussed this but not sure where we landed on having this turned on or not. Could it be an option so a user could opt in to having their run crash if states and integrated fluxes start to diverge?

**Review by JessicaNeedham — APPROVED** [review]
Thanks Ryan this looks great! My simulations with this branch merged also show very tiny divergence between integrated fluxes and stocks.

File: biogeochem/EDPhysiologyMod.F90 [hunk 3]
```diff
@@ -2878,9 +2888,9 @@ subroutine CWDInput( currentSite, currentPatch, litt, bc_in)
        !        about double counting.
        ! ---------------------------------------------------------------------------------
 
-       flux_diags%leaf_litter_input(pft) = &
-            flux_diags%leaf_litter_input(pft) +  &
-            leaf_m_turnover * currentCohort%n
+       elflux_diags%leaf_litter_input(pft) = &
```
[the remark this answers is not in the document]
Comment (rgknox) [reply biogeochem/EDPhysiologyMod.F90]: renamed surf_fine_litter_input

File: main/ChecksBalancesMod.F90 [hunk 4]
```diff
@@ -240,7 +243,134 @@ subroutine CheckLitterPools(currentSite,bc_in)
     return
   end subroutine CheckLitterPools
 
+  ! ==================================================================
```
[the remark this answers is not in the document]
Comment (rgknox) [reply main/ChecksBalancesMod.F90:246]: added descriptive comment

File: main/FatesHistoryInterfaceMod.F90 [hunk 5]
```diff
@@ -6433,12 +6422,6 @@ subroutine define_history_vars(this, initialize_variables)
                upfreq=group_nflx_simple, ivar=ivar, initialize=initialize_variables,              &
                index = ih_nefflux_si)
 
-          call this%set_history_var(vname='FATES_NDEMAND', units='kg m-2 s-1',      &
```
[the remark this answers is not in the document]
Comment (rgknox) [reply main/FatesHistoryInterfaceMod.F90]: added them back in

File: main/ChecksBalancesMod.F90 (hunk 2 above)
Comment (rgknox) [reply main/ChecksBalancesMod.F90:345]: I added an issue here: https://github.com/NGEET/fates/issues/1264

File: main/ChecksBalancesMod.F90 (hunk 1 above)
Comment (rgknox) [reply main/ChecksBalancesMod.F90]: fixed

**Comment by rgknox** [comment]
@mpaiao and @JessicaNeedham I addressed your comments. @mpaiao I changed the name of that variable from leaf_* to surf_fine_* to be more general and include seed.  @JessicaNeedham I added back in those history variables and also created an issue about graceful failure.

**Comment by rgknox** [comment]
All tests b4b except for the specific output variables that were changed in this PR with intent.
/glade/derecho/scratch/rgknox/tests_1017-183514de

**Pull request merged**